Google Sheets: migrate actions to dedicated pages (#46252)
home-assistant/home-assistant.github.io: source/_actions/google_sheets.get_sheet.markdown

---
title: "Get sheet"
action: google_sheets.get_sheet
domain: google_sheets
description: "Retrieves rows of data from a Google Sheets document."
related_actions:
  - google_sheets.append_sheet
---

Use this action to retrieve rows of data from a Google Sheets document, for example, to read back values you stored earlier. This action returns [response data](/docs/scripts/perform-actions/#use-templates-to-handle-response-data) that you can use in the rest of your automation or script.

{% include actions/ui_header.md %}

To get data from an automation or a script:

1. Go to {% my automations title="**Settings** > **Automations & scenes**" %}.
2. Open an existing automation or script, or select **Create automation** > **Create new automation**.
3. If you're setting up a new automation, add a trigger in the **When** section. Scripts don't need a trigger. They run when something else calls them.
4. In the **Then do** section, select **Add action**.
5. From the search box, search for and select **Google Sheets: Get sheet**.

> 6. Select the **Config entry** for the document you want to read from.

Reviewer: ```suggestion
6. Under **Integration**, select the **Sheet** you want to read from.
```

> 7. Optionally, set the **Worksheet**.

Reviewer: ```suggestion
7. Optionally, set a name for the **Worksheet**. Defaults to the first one in the document.
```

8. Set the number of **Rows** to return.

> 9. Select **Save**.

Reviewer: ```suggestion
8. Set the number of **Rows** to return.
9. In the **Response variable** field, enter a name to store the data in, such as `sheet_data`.
10. Select **Save**.``
